Backups segfault on ARMv7

I build restic with Jenkins from Github master as and when commits are pushed.
Until this time I have not had any issues with it backing files up... until now ;-(

<!--

Welcome! - We kindly ask that you:

  1. Fill out the issue template below - not doing so needs a good reason.
  2. Use the forum if you have a question rather than a bug or feature request.

The forum is at: https://forum.restic.net

NOTE: Not filling out the issue template needs a good reason, as otherwise it
may take a lot longer to find the problem, not to mention it can take up a lot
more time which can otherwise be spent on development. Please also take the
time to help us debug the issue by collecting relevant information, even if
it doesn't seem to be relevant to you. Thanks!

The forum is a better place for questions about restic or general suggestions
and topics, e.g. usage or documentation questions! This issue tracker is mainly
for tracking bugs and feature requests directly relating to the development of
the software itself, rather than the project.

Thanks for understanding, and for contributing to the project!

-->


Output of `restic version`
--------------------------
```
# restic version
debug log file debug.log
debug enabled
restic 0.9.6 (v0.9.6-135-ga307797c) compiled with go1.14 on linux/arm
```

How did you run restic exactly?
-------------------------------
```
export B2_ACCOUNT_ID="XXXX"
export B2_ACCOUNT_KEY="XXXX"
export RESTIC_REPOSITORY="b2:restic-$(hostname | md5sum | awk '{print $1}')-$(hostname)"
export RESTIC_PASSWORD_FILE="/root/.restic"
export RESTIC_CACHE_DIR="/data/.restic"
restic init
restic backup /etc
restic backup /etc
```

The 1st backup of /etc will work ok, but the 2nd backup is when things segfault.

<!--
This section should include at least:

 * The complete command line and any environment variables you used to
   configure restic's backend access. Make sure to replace sensitive values!

 * The output of the commands, what restic prints gives may give us much
   information to diagnose the problem!
-->

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
Backblaze B2


Expected behavior
-----------------
Restic should backup correctly.
My last successful backup on this host was on the 27th Feb 2020.
(I have another linux host that bacups hourly and is working just fine.)

<!--
Describe what you'd like restic to do differently.
-->

Actual behavior
---------------
It dies with a Segmentation fault :-(
```
# restic backup /etc
debug log file debug.log
debug enabled
repository 4357d291 opened successfully, password is correct
Segmentation fault files 4.264 MiB, total 941 files 6.448 MiB, 0 errors ETA 0:47
root@ReadyNAS:~/test#
/etc/wgetrc
```
Each time I run a backup it crashes but It doesn't stop at the same point in time or file.
<!--
In this section, please try to concentrate on observations, so only describe
what you observed directly.
-->

Steps to reproduce the behavior
-------------------------------
I'm just running a backup (well, attempting to)

<!--
The more time you spend describing an easy way to reproduce the behavior (if
this is possible), the easier it is for the project developers to fix it!
-->

Do you have any idea what may have caused this?
-----------------------------------------------
I am using a [ReadyNAS Duo](https://www.netgear.com/support/product/RND2000v1_(ReadyNAS_Duo_v1).aspx) to run this backup on. It has 512MB of RAM, and lots of swap space allocated that it never actually uses.


Do you have an idea how to solve the issue?
-------------------------------------------
Restic backs up just fine on a Linux x86_64 system with 16GB RAM. Maybe its a memory thing, but its odd that its only just stopped working.
I have deleted my repo and started again with a fresh one at Backblaze so I don't believe data to be corrupted.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
Yes, and no.

<!--
Answering this question is not required, but if you have anything positive to share, please do so here!
Sometimes we get tired of reading bug reports all day and a little positive end note does wonders.
Idea by Joey Hess, https://joeyh.name/blog/entry/two_holiday_stories/
-->

I also recently experienced restic segfaulting on armv7l:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x353e8c]

goroutine 208 [running]:
github.com/restic/restic/internal/repository.(*Index).ID(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        internal/repository/index.go:377 +0x3c
github.com/restic/restic/internal/repository.(*Repository).LoadIndex.func5(0x4, 0x8c5e50)
        internal/repository/repository.go:467 +0x88
golang.org/x/sync/errgroup.(*Group).Go.func1(0x1348f90, 0x1336ef0)
        vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x48
created by golang.org/x/sync/errgroup.(*Group).Go
        vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x50
```

`restic version`: `restic 0.9.5 compiled with go1.12.9 on linux/arm`

I realize this was built with an older go, and this *seems* to have been a one time thing so I'm not too concerned. I just thought I should include this here for completeness.

--- a/go.mod
+++ b/go.mod
@@ -33,7 +33,7 @@ require (
 	github.com/spf13/pflag v1.0.3
 	github.com/stretchr/testify v1.3.0 // indirect
 	go.opencensus.io v0.20.2 // indirect
-	golang.org/x/crypto v0.0.0-20190513172903-22d7a77e9e5f
+	golang.org/x/crypto v0.0.0-20200427165652-729f1e841bcc
 	golang.org/x/net v0.0.0-20190522155817-f3200d17e092
 	golang.org/x/oauth2 v0.0.0-20190402181905-9f3314589c9a
 	golang.org/x/sync v0.0.0-20190423024810-112230192c58

